src/Riok.Mapperly/Configuration/MergedMapperConfiguration.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Descriptors;
+using Riok.Mapperly.Helpers;
+
+namespace Riok.Mapperly.Configuration;
+
+public class MergedMapperConfiguration
+{
+    private readonly MappingConfiguration _defaultConfiguration;
+    private readonly AttributeDataAccessor _dataAccessor;
+
+    public MergedMapperConfiguration(SymbolAccessor symbolAccessor, ISymbol mapperSymbol)
+    {
+        _dataAccessor = new AttributeDataAccessor(symbolAccessor);
+        var mapperConfiguration = _dataAccessor.AccessSingle<MapperAttribute, MapperConfiguration>(mapperSymbol);
+        var defaultMapperConfiguration = _dataAccessor.AccessAssemblyFirstOrDefault<DefaultMapperAttribute, MapperConfiguration>();
```

Comment:
This reads the assembly configuration for each mapper class which could be inefficient for large code bases. Instead we should try to read it once for all mapper classes. This should work similar to how we extract the class declarations in the mapper generator. Eg. for roslyn 4.4 this could look like:
```c#
public static IncrementalValuesProvider<CompilationUnitSyntax> GetAssemblyDefaultMappers(IncrementalGeneratorInitializationContext context)
    {
        return context.SyntaxProvider.ForAttributeWithMetadataName(
            "Riok.Mapperly.Abstractions.DefaultMapperAttribute",
            static (s, _) => s is CompilationUnitSyntax,
            static (ctx, _) => ctx.TargetNode as CompilationUnitSyntax);
}
```

Reply:
Used a different way. Accessed the assembly attributes with the `Microsoft.CodeAnalysis.Compilation` which is already provided through the context.